src/unix/fs.c

```diff
@@ -929,6 +929,10 @@ static ssize_t uv__fs_copyfile(uv_fs_t* req) {
     }
   }
 
+  /* If -1 is returned, errno is used. */
+  if (result != 0)
+    errno = result;
+
```

Comment:
Not wrong, not exactly, but should probably be along these lines:
```c
if (result == 0)
  return 0;

errno = result;
return -1;
```

Reply:
Updated. I had to add `errno = UV__ERR(result);`. Otherwise there were some issues regarding the sign on `errno` because `UV__ERR()` is called again in `uv__fs_work()`.

Reply:
Right you are.